src/Symfony/Component/Validator/Constraints/ExpressionValidator.php

```diff
@@ -27,15 +27,16 @@ class ExpressionValidator extends ConstraintValidator
 
     public function __construct(/*ExpressionLanguage */$expressionLanguage = null)
     {
-        if (!$expressionLanguage instanceof ExpressionLanguage) {
-            if (null !== $expressionLanguage) {
-                @trigger_error(sprintf('The "%s" first argument must be an instance of "%s" or null since 4.4. "%s" given', __METHOD__, ExpressionLanguage::class, \is_object($expressionLanguage) ? \get_class($expressionLanguage) : \gettype($expressionLanguage)), E_USER_DEPRECATED);
-            }
+        if (\func_num_args() > 1) {
+            @trigger_error(sprintf('The "%s" instance should be passed as "%s" first argument instead of second argument since 4.4.', ExpressionLanguage::class, __METHOD__), E_USER_DEPRECATED);
+
+            $expressionLanguage = func_get_arg(1);
 
-            if (\func_num_args() > 1 && func_get_arg(1) instanceof ExpressionLanguage) {
-                @trigger_error(sprintf('The "%s" instance should be passed as "%s" first argument instead of second argument since 4.4.', ExpressionLanguage::class, __METHOD__), E_USER_DEPRECATED);
-                $expressionLanguage = func_get_arg(1);
+            if (null !== $expressionLanguage && !$expressionLanguage instanceof ExpressionLanguage) {
+                throw new \TypeError(sprintf('Argument 2 passed to %s() must be an instance of %s, %s given.', __METHOD__, ExpressionLanguage::class, \is_object($expressionLanguage) ? \get_class($expressionLanguage) : \gettype($expressionLanguage)));
```

Comment:
Should we hint directly in the exception message to pass it as the first argument by the same way?

Reply:
Sounds reasonable, any good idea how to phrase that?

Reply:
```diff
- Argument 2 passed to %s() must be an instance of %s, %s given.
+ Argument 2 passed to %s() must be an instance of %s or null, %s given. Since 4.4, passing it as the second argument is deprecated and will trigger a deprecation. Pass it as first argument instead.
```

?

Reply:
that's good 👍